invertedai/utils.py

```diff
@@ -243,13 +245,20 @@ def request(
         self, model: str, params: Optional[dict] = None, data: Optional[dict] = None
     ):
         method, relative_path = iai.model_resources[model]
+        
+        if self._debug_logger is not None:
+            self._debug_logger.append_request(model,json.dumps(data))
```

Comment:
So you are passing a json string here into the append_request() but this function itself is accepting a dictionary as well and later dumps it again ?

Reply:
Unfortunately this is done to align with the C++ SDK where it writes requests/responses as strings which are appended into the C++ equivalent of lists and then the whole thing needs to be written to a JSON file.

Reply:
Also to clarify, what is being passed to these functions is a Dict object which is then dumped into a json string.

Reply:
I'll add data types to the function parameters to clarify this.